---
title: "Lab10_KatieTumang"
author: "Katie Tumang"
date: "11/28/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(vcfR)
```

```{r}
data(vcfR_example)
vcf
```

```{r}
strwrap(vcf@meta[1:7])
```

```{r}
queryMETA(vcf)
```

```{r}
queryMETA(vcf, element = 'FORMAT=<ID=DP')
```

```{r}
head(getFIX(vcf))
```

```{r}
vcf@gt[1:6, 1:4]
```

```{r}
vcf <- read.vcfR("pinfsc50_filtered.vcf")
```

```{r}
head(vcf)
```

```{r}
write.vcf(vcf, "myVCFdata_filtered.vcf.gz")
```

### Exercise 1

#### 1. How would we find more information about read.vcfR()?

```{r}
?read.vcfR
```

#### 2. How would we learn what the acronym “AD” stands for?

```{r}
queryMETA(vcf, element = 'AD')
```

#### 3. We used the head() function to view the first few lines of fix data. How would we view the last few lines of fix data?

```{r}
tail(vcf@fix)
```

#### 4. There is a column in the fix portion of the data called QUAL. It is not defined in the meta portion of the data because it is defined in the VCF specification. It stands for ‘quality’. Does QUAL appear useful to us? Why or why not?

```{r}
plot(vcf)
```

##### QUAL seems to be useful, as it defines the quality of the data -- if the data is low quality, then it likely is not accurate/reliable and this should be taken into account when using the data.

#### 5.  How would we query the sample names?

```{r}
colnames(vcf@gt)
```

## Part II - Analysis of Genome Data

```{r}
library('vcfR')
vcf <- read.vcfR("pinfsc50_filtered.vcf")
```

```{r}
vcf
```

### Converting VCF data to a genlight object

```{r}
x <- vcfR2genlight(vcf)
```

```{r}
x
```

```{r}
# vcfR check
gt <- extract.gt(vcf, element = "GT")
gt[c(2,6,18), 1:3]
```

```{r}
# genlight check
t(as.matrix(x))[c(1,5,17), 1:3]
```

```{r}
library(adegenet)
```

```{r}
pop(x) <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
popNames(x)
```

```{r}
ploidy(x) <- 2
```

### Distance matrices

```{r}
x.dist <- dist(x)
```

```{r}
x.dist <- poppr::bitwise.dist(x)
```

### Creating chromR objects

```{r}
install.packages('pinfsc50', repos = "http://cran.us.r-project.org")
library(vcfR)
```

```{r}
# find the files
vcf_file <- system.file("extdata", "pinf_sc50.vcf.gz", package = "pinfsc50")
dna_file <- system.file("extdata", "pinf_sc50.fasta", package = "pinfsc50")
gff_file <- system.file("extdata", "pinf_sc50.gff", package = "pinfsc50")
```

```{r}
# input the files
vcf <- read.vcfR(vcf_file, verbose = FALSE)
dna <- ape::read.dna(dna_file, format = "fasta")
gff <- read.table(gff_file, sep="\t", quote="")
```

```{r}
# create a chromR object
chrom <- create.chromR(name="Supercontig", vcf=vcf, seq=dna, ann=gff, verbose=TRUE)
```

```{r}
chrom
```

```{r}
plot(chrom)
```

```{r}
chromoqc(chrom, dp.alpha = 66)
```

### Processing chromR objects

```{r}
chrom <- proc.chromR(chrom, verbose = TRUE)
```

```{r}
plot(chrom)
```

```{r}
chromoqc(chrom, dp.alpha = 66)
```

```{r}
vcf <- read.vcfR("pinfsc50_filtered.vcf", verbose = FALSE)
chrom <- create.chromR(name="Supercontig", vcf=vcf, seq=dna, ann=gff, verbose=FALSE)
chrom <- proc.chromR(chrom, verbose = FALSE)
chromoqc(chrom, dp.alpha = 66)
```

### Tabular summaries

```{r}
head(chrom@var.info)
```

```{r}
head(chrom@win.info)
```

### Genetic differentiation

```{r}
library(vcfR)
data(vcfR_example)
pop <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
myDiff <- genetic_diff(vcf, pops = pop, method = 'nei')
knitr::kable(head(myDiff[,1:15]))
```

```{r}
knitr::kable(head(myDiff[,16:19]))
```

```{r}
knitr::kable(round(colMeans(myDiff[,c(3:9,16,19)], na.rm = TRUE), digits = 3))
```

##### Another way to summarize using violin plots

```{r}
library(reshape2)
library(ggplot2)

dpf <- melt(myDiff[,c(3:8,19)], varnames=c('Index', 'Sample'), value.name = 'Depth', na.rm=TRUE)
```

```{r}
p <- ggplot(dpf, aes(x=variable, y=Depth)) + geom_violin(fill="#2ca25f", adjust = 1.2)
p <- p + xlab("")
p <- p + ylab("")
p <- p + theme_bw()
p
```

#### Exercise 2

##### 1. You actually have everything you need to make a Manhattan plot. Can you figure out how to plot G′ST (y-axis) by genomic position (POS)?

```{r}
plot(getPOS(vcf), myDiff$Gprimest, col="#6BB5BE", xlab = "", ylab = "", xaxt="n", ylim = c(0, 1))
title(main = "Genomic position by G'ST")
axis(side = 1, at = seq(0, 1e5, by = 1e4), labels = seq(0, 100, by = 10))
title(xlab='genomic position (Kbp)')
title(ylab = expression(italic("G'"["ST"])))
```


##### 2. This Manhattan plot should look a bit unusual. Can you think of anything that may be wrong with this analysis?

###### There is a very small sample size -- normally, Manhattan plots are used when there is a large amount of data (e.g., whole genomes.)The small population is visualized below.

```{r}
table(pop)
```

##### 3. Can you figure out how to zoom in on a particular region of a chromosome in chromoqc()?

```{r}
chromoqc(chrom, dp.alpha = 66, xlim = c(3e07, 3e09))
```

##### 4. Can you use the function queryMETA() to look for other data in your file that may be of interest?

```{r}
queryMETA(vcf)
```

